compressed_depth_image_transport/src/compressed_depth_publisher.cpp

```diff
@@ -115,8 +115,10 @@ void CompressedDepthPublisher::advertiseImpl(
 
   // Declare Parameters
   uint ns_len = node->get_effective_namespace().length();
-  std::string param_base_name = base_topic.substr(ns_len);
-  std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
+  if (base_topic.length() >= ns_len) {
+    std::string param_base_name = base_topic.substr(ns_len);
```

Comment:
Did you already try this patch ? There is a compilation error 

```bash
14:50:25 /tmp/ws/src/image_transport_plugins/zstd_image_transport/src/zstd_publisher.cpp: In member function ‘virtual void zstd_image_transport::ZstdPublisher::advertiseImpl(rclcpp::Node*, const string&, rmw_qos_profile_t, rclcpp::PublisherOptions)’:
14:50:25 /tmp/ws/src/image_transport_plugins/zstd_image_transport/src/zstd_publisher.cpp:93:22: error: ‘param_base_name’ was not declared in this scope
14:50:25    93 |     declareParameter(param_base_name, pd);
```

`param_base_name` is defined inside the `if` scope but this variable is used in the outer scope

Reply:
I haven't tried it, sorry for that. I found this issue resulting in exception in my code, and I adopted a workaround, and then I forgot to compile and test this patch. I'll fix it soon.